modules/runtime/nuxeo-runtime-jtajca/src/main/java/org/nuxeo/runtime/transaction/TransactionHelper.java

```diff
@@ -216,6 +216,36 @@ public static boolean isTransactionTimedOut() {
         }
     }
 
+
+    /**
+     * Gets the transaction TTL.
+     *
+     * @return the time to live in millis or {@code -1} when outside of a transaction.
+     * @since 11.5
+     */
+    public static long getTransactionTimeToLiveMs() {
+        TransactionManager tm = NuxeoContainer.getTransactionManager();
+        if (tm == null) {
+            return -1;
+        }
+        try {
+            Transaction tx = tm.getTransaction();
+            if (tx == null || tx.getStatus() != Status.STATUS_ACTIVE) {
+                return -1;
+            }
+            if (tx instanceof org.apache.geronimo.transaction.manager.TransactionImpl) {
+                // Geronimo Transaction Manager
+                return (Long) GERONIMO_TRANSACTION_TIMEOUT_FIELD.get(tx);
```

Comment:
You must substract `System.currentTimeMillis()` to get a TTL, as Geronimo stores the expiration time of the transaction.
Also we could normalize zero and negative values (tx expired) to `0`. Behavior when tx is expired should be documented in the Javadoc.